java/src/main/java/ai/rapids/cudf/JCudfSerialization.java

```diff
@@ -720,6 +720,9 @@ private static SerializedTableHeader calcHeader(ColumnBufferProvider[] columns,
                                                   int numRows) {
     DType[] types = new DType[columns.length];
     long[] nullCount = new long[columns.length];
+    if (columns.length == 0) {
```

Comment:
This should be before the arrays are heap-allocated, or alternatively this check shouldn't be here and instead `getSlicedSerializedDataSizeInBytes` is updated to handle an empty columns array.